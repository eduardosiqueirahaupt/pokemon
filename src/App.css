.app-container {
  height: 100%;
  position: absolute;
  width: 100%;
  overflow-x: hidden;
  overflow-y: auto;
}

::-webkit-scrollbar {
    width: 10px;
  }
  
  /* Track */
  ::-webkit-scrollbar-track {
    box-shadow: inset 0 0 5px grey; 
    border-radius: 10px;
  }
   
  /* Handle */
  ::-webkit-scrollbar-thumb {
    background: rgba(253,219,45,1); 
    border-radius: 10px;
  }
  
  /* Handle on hover */
  ::-webkit-scrollbar-thumb:hover {
    background: rgba(253,219,45,1); 
  }