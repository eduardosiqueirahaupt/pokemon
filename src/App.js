import React, { useEffect } from "react";
import { Route, Routes } from "react-router-dom";
import Header from "./components/header";
import Home from "./pages/home";
import Favorites from "./pages/favorites";
import './App.css'

function App() {
  const initPokemonsFav = () => {
    if (!localStorage.getItem("FavPokemons")) {
      localStorage.setItem("FavPokemons", JSON.stringify([]))
    }
  }

  useEffect(() => {
    initPokemonsFav()
  }, [])

  return (
    <div className="app-container">
      <Header />
      <Routes>
        <Route path="/" exact element={<Home />} />
        <Route path="/favorites" exact element={<Favorites />} />
      </Routes>
    </div>
  );
}

export default App;
