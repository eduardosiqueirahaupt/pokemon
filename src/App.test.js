import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import App from './App'

const renderComponent = () => (
  render(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  )
)

describe("App", () => {
  it("Renderizar App", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  })
})