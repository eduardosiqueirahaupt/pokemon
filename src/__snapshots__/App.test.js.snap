// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renderizar App 1`] = `
<div>
  <div
    class="app-container"
  >
    <div
      class="sc-bczRLJ hmunFc"
    >
      <img
        alt="pokemon-logo"
        class="sc-hKMtZM BPKtS"
        src="pokemon-logo.png"
      />
      <div
        class="sc-gsnTZi geleqQ"
      >
        <a
          href="/"
        >
          <span
            class="sc-dkzDqf cRnlOR"
          >
            Inicio
          </span>
        </a>
        <a
          href="/favorites"
        >
          <span
            class="sc-dkzDqf cRnlOR"
          >
            Favoritos
          </span>
        </a>
      </div>
    </div>
    <div
      class="sc-jqUVSM klFaXP"
    >
      <div
        class="sc-ftvSup cSmjlD"
      >
        <input
          class="sc-papXJ bIXmnd"
          data-testid="input-filter"
          name="search"
          placeholder="Buscar..."
          type="text"
          value=""
        />
      </div>
      <div
        class="sc-kDDrLX eNTzbW"
      >
        <div
          class="sc-iqcoie fTAtuW"
        >
          <div
            class="sc-crXcEl hxIgXv"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
